From 9ea6650d3766c897ef85a3df15fdd9abcc9ee858 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Mon, 26 Nov 2018 00:37:45 -0800
Subject: [PATCH] fbsd: Correct signdedness in setmsr callback

src/firmware/fbsd.c:693:7: error: implicit conversion changes signedness: 'unsigned int' to 'int' [-Werror,-Wsign-conversion]
        case MSR_EFER:
             ^~~~~~~~
In file included from src/firmware/fbsd.c:80:
include/xhyve/support/specialreg.h:771:18: note: expanded from macro 'MSR_EFER'
                        ^~~~~~~~~~

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 include/xhyve/firmware/fbsd.h | 2 +-
 src/firmware/fbsd.c           | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/xhyve/firmware/fbsd.h b/include/xhyve/firmware/fbsd.h
index 30d4388..c1f10ef 100644
--- a/include/xhyve/firmware/fbsd.h
+++ b/include/xhyve/firmware/fbsd.h
@@ -66,7 +66,7 @@ struct loader_callbacks {
 	/* Set a guest register value */
 	void (*setreg)(void *arg, int, uint64_t);
 	/* Set a guest MSR value */
-	void (*setmsr)(void *arg, int, uint64_t);
+	void (*setmsr)(void *arg, u_int, uint64_t);
 	/* Set a guest CR value */
 	void (*setcr)(void *arg, int, uint64_t);
 	/* Set the guest GDT address */
diff --git a/src/firmware/fbsd.c b/src/firmware/fbsd.c
index 6a2d2bc..19d1936 100644
--- a/src/firmware/fbsd.c
+++ b/src/firmware/fbsd.c
@@ -682,7 +682,7 @@ cb_setreg(UNUSED void *arg, int r, uint64_t v)
 }
 
 static void
-cb_setmsr(UNUSED void *arg, int r, uint64_t v)
+cb_setmsr(UNUSED void *arg, u_int r, uint64_t v)
 {
 	int error;
 	enum vm_reg_name vmreg;
